flight/Modules/Actuator/actuator.c

```diff
@@ -191,76 +214,34 @@ static void actuatorTask(void* parameters)
 	{
 		PIOS_WDG_UpdateFlag(PIOS_WDG_ACTUATOR);
 
-		// Wait until the ActuatorDesired object is updated
-		bool rc = PIOS_Queue_Receive(queue, &ev, FAILSAFE_TIMEOUT_MS);
+		bool rc=false;
 
-		/* Process settings updated events even in timeout case so we always act on the latest settings */
-		if (actuator_settings_updated) {
-			actuator_settings_updated = false;
-			ActuatorSettingsGet (&actuatorSettings);
-			actuator_update_rate_if_changed (&actuatorSettings, false);
+		// If it's our first iteration, fall down to getting the
+		// config and enabling failsafe.
+		if (!first) {
```

Comment:
Okay, this still isn't making sense to me. Wrapping up those three lines of code into a function shouldn't confuse the compiler. I get the feeling we're having a bit of a comms barf right now, so I'll prepare a gist with the full proposed solution.